ReverseTests » testCrankyEvaluateBlockWithoutNulls failure with certain seeds

### CI Link

https://gradle-enterprise.elastic.co/s/557gksjia7quy/tests/task/:x-pack:plugin:esql:test/details/org.elasticsearch.xpack.esql.expression.function.scalar.string.ReverseTests/testCrankyEvaluateBlockWithoutNulls%20%7BTestCase=%3Clong%20unicode%20KEYWORD%3E%7D?expanded-stacktrace=WyIwIl0&top-execution=1

### Repro line

./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.scalar.string.ReverseTests.testCrankyEvaluateBlockWithoutNulls {TestCase=<long unicode KEYWORD>}" -Dtests.seed=6C140CED5816D457 -Dtests.locale=bho-IN -Dtests.timezone=Europe/Kiev -Druntime.java=22

### Does it reproduce?

Yes

### Applicable branches

8.x, main

### Failure history

_No response_

### Failure excerpt

```
java.lang.AssertionError: 
Expected: <[61 5e 1 70 18 8 5f 6 e 77 14 77 4d 22 26 4c 63 2 1e 5a 58 75 2c 2b 16 5d 3 e 4f b 76 51 40 1d 39 20 2c 27 52 15 8 1 58 9 3a 5d 20 3f 3c 33 22 4c 1b 25 4 e 3f 2f 66 63 43 70 53 c 1c 5a 78 27 8 7c 43 3d f 2 12 62 79 71 b 32 45 1b 2 60 2 0 19 39 2 3 32 1c 14 6b 11 a 4f 4f c 37 1e 36 37 67 12 3c 6e 17 3e 10 2d 1f 7f 4e 7e a 9 26 5 26 49 2a 7f e 76 67 7c 47 2 1d 73 0 3c 5a 3a 3e 4d 12 34 3f 1a 10 17 12 27 6d 4e 40 6a 1 5b 77 7c 5e 4c 24 79 34 3c 48 15 c 33 27 6e 32 3c 44 7f 56 3c 12 27 31 f 15 1c 7d 31 24 3c 37 5e 20 2c 32 43 31 20 7b 62 3d 7c 7d 44 6 68 35 32 29 11 34 22 1c 63 4 7e 76 73 4d 44 4d 42 5a 39 5a 78 60 d 1b 61 6a 6c 4e 24 1b 13 66 79 1b 18 50 3 52 2d 4b 52 f 1e 21 d 73 0 47 11 1b 59 43 7 43 66 5f 1e 36 31 1a 4b 14 1 6e 3 2b 4c 72 22 19 57 15 20 15 5f 36 39 d 54 37 46 41 4e 6a 34 38 2d 2a 7b 73 b 3d 3e 73 41 7e 47 10 67 21 7e 6f 2f 30 3e 72 3c 50 79 14 f 41 5c 7b c 44 17 24 2f 9 6e 16 53 23 c 15 3a 56 54 1d e 6e 21 59 51 5e 4d 73 5c 7e 47 32 5a 9 66 5 28 4e 6f 11 7f 45 e 29 41 46 66 69 5 47 4b 3f 42 7 65 6f 3 14 4c 1 8 d 55 63 62 14 50 2f 42 8 4d 2a 2f 6d 37 19 4a 3c 65 7b 4f 3 41 1a 7e 64 20 12 7b 28 2b 49 6e 18 7e d 6 e 20 54 11 5a 67 70 1f 1 79 53 22 46 76 28 35 3a 65 2 4 74 42 47 18 a 43 3d 79 60 6a 73 23 4e 56 2 1 56 65 62 34 a 25 2d 4a 74 39 15 5f 59 48 11 3c 1d 56 3e 39 37 7b 2 72 30 4e 51 0 45 6 48 21 b 54 26 32 64 20 43 2f 55 4e 10 61 4c 6f 73 4 57 5e 23 51 f 56 b 25 62 17 48 4c e 19 62 10 2c 29 1f 60 51 1f 3c 3b 7d 3f 13 64 53 5d 3b 71 3f 1e 2b 70 22 1b 4 1a 9 d 17 5b f 50 3e 5e 3e 52 76 35 7f 62 1b 2b 35 46 50 26 3a 47 68 7d 1b 7e a 2b 58 11 46 6b 4 2a 70 3 7c 2b 17 5e 33 7 5d 32 5d 62 7 2d 10 3 25 74 1c 8 2d 4d 37 5c 31 2e 0 6f c 6f 0 59 7 18 0 17 1f 78 76 47 5b 5a 1c c 4c 5 1e 1c 61 46 53 11 52 36 1d 41 5b 39 14 6b b 3b 6f 29 8 4e 2c 1e 37 6e 6a 23 7c 4b 59 27 5a 2d 72 68 3d 42 1c 42 17 1a 3b 15 16 49 1d 20 69 7f 5b 32 79 39 7c 6b 54 2b 74 2 69 15 7f 6a 6 38 42 8 70 11 2 12 3e 12 65 3d 6 3a 6d 4e 64 24 3b 34 50 a b 38 38 61 20 4c 29 6f 29 33 a e 13 24 46 20 2a 7c c 37 b 2a 67 7d 6a 9 e 69 46 6f 7e e 6 3d c 6 73 5a 0 71 51 7d 44 7c f 69 1c 31 6b 39 2e 26 31 4b 50 7c d 62 13 48 51 e 62 2c 55 48 a 25 2a 2a 7d 31 33 4a 22 77 3e 65 7e 61 3e 3c 26 64 14 10 2b 62 36 7d e 2 32 70 32 15 d 7b 4d 22 38 1c 1f 3 24 7f 7 57 3d 6b 50 27 53 5c 51 13 43 53 6c 22 63 2f 24 6f 61 8 57 64 43 40 5c 11 54 4b 75 38 1c 7 13 62 46 20 29 44 4f 54 50 40 7d 32 1d 42 f 6f 57 4b 39 69 1f 38 1d 54 21 1b 51 30 60 49 2d 70 1b 1e 23 58 4d 5c 7e 47 43 1d 74 19 4c 2c 23 3c 20 70 65 77 40 53 56 25 74 33 50 51 68 75 53 e 43 2b 7f 63 18 42 71 67 c 5c 19 7c 22 3 3d 21 60 d 76 a 30 31 3e 3c 13 6e 4b 7d 7d 53 71 2d 72 71 65 38 7b 10 7b 19 53 3f 46 59 53 4e 2c 23 2a 3c 1a 1d 20 10 67 14 5a 52 5a 10 4d 10 40 57 1b 8 26 63 6d 3d 25 23 18 6d 1e 0 58 60 39 51 58 6a 77 24 13 22 38 1f 4d 57 37 6e 5d 4c 6d 38 6f 46 7f 4b 7f 6a 7f 20 12 77 20 63 2e 1d 10 9 23 42 13 0 55 2a 79 1e 73 1c 71 1b 1c 34 20 6d 6 23 9 4f 22 7f 28 6b 41 3 32 19 75 53 27 57 66 31 59 8 1 73 6 5d 6 1b 1b 2e 42 53 26 59 38 46 7c 39 27 6a 31 2e 4a 1a 23 12 63 6c 1e 55 66 45 1b 5c 15 4d 11 58 0 41 7d 4e 6 67 47 28 18 3b 68 5f 13 31 59 7 16 63 4e 21 75 4f 35 39 69 52 3b 4 4c 39 6b 32 20 33 75 22 23 14 50 e 44 16 48 2c 46 26 d 7 34 4b 3a 3c 42 49 4c d 22 47 8 3 5e 4b 6a 63 3b f 71 34 58 5a 1e 74 32 22 5e 19 6d 6e 3a 60 61 24 23 30 56 63 75 6 5 2a 42 29 18 5 56 5c 68 1e 1a 5f 1c 41 2 60 68 76 43 43 27 12 50 65 79 6c 54 25 51 72 35 57 44 11 54 54 30 70 57 5e 2a 6e 5b 5d 37 70 6b 5 b 65 12 41 3d 4c f 3e 2d 39 a 4e 26 28 7f 3b 3a 58 30 77 52 51 14 7d 0 54 2 66 4c 46 30 14 61 46 1a 18 52 2 9 2b 1b c 3f 59 28 7e 33 31 32 7c 66 70 39 5c 19 67 32 11 42 7d 73 22 79 6e 44 11 66 29 2c 7 4b 7e 5c f 37 64 16 33 8 7d 78 77 74 12 b 50 7 34 1 8 7b 79 34 21 36 56 79 48 58 6 18 7a 70 41 31 3d 3a 15 58 7 20 7f 24 54 13 51 1c 58 3e 9 3f 2d 42 66 13 32 54 17 58 f 37 51 7f 6a 2 3a e 4 47 75 31 68 71 60 59 1e 1e 46 40 50 6c 7a 3a 1d 76 21 6f 3e 34 11 55 30 42 64 11 4c 5f 30 70 17 3 28 26 5b 33 7d 68 4 19 75 6e 2 69 66 7c 5b 79 76 69 4b 6b 50 0 b 7f 57 14 25 5c 7e c 22 8 32 0 28 36 31 27 50 f 45 29 7b 41 66 1c 1f 64 2f 2d 28 26 29 1c 5d 63 6d 7b 2b 4a 6f 4f 66 25 0 4c 16 7e 37 e 62 2c 5c 73 62 52 7f 4e 2a 30 4b 30 39 12 6b 27 77 51 1a 39 79 64 68 42 60 60 2d 5a 59 6c 4f 11 63 1f 53 66 3c 36 43 39 42 78 5b 7 24 46 7c 5b 7b 71 40 75 7a 3c 55 13 27 62 74 37 49 65 12 24 66 3 8 12 6c 64 11 56 41 7a 38 4c 1b f 67 25 14 74 1f 55 1b 4f 9 e 60 4f 46 f 42 20 4b 9 51 48 43 42 71 4a 4d 7a 23 30 53 44 41 1 2d 7d 4d 58 15 33 2d 30 57 56 41 28 a 8 36 5b 31 75 38 61 1c 6c 5c 49 11 75 6c 18 f 48 6a b 37 2d 4b 30 5e 19 21 59 7c 7a 16 18 b 4f 3f 76 6d 28 16 33 55 57 31 16 1d 7 51 34 48 7e 6 52 39 6c 34 d 44 5e 65 2e 41 21 40 7b 5a 78 39 30 76 5b 12 11 3d 59 51 18 23 5e 2f 44 57 9 6b 2a 17 d a 79 4 35 7b 38 6a 7a 3e 2 3 18 d 48 7b 40 35 63 76 3f 41 4d 66 0 a 7d 1 59 58 5 65 30 1 77 2f 4 5d 62 5d 6b 39 6a 18 5e 4b 67 0 11 0 65 6b 4b 36 1e 6d 66 0 78 71 3e 22 5 17 74 41 9 1a 73 50 3c 12 48 39 66 7e 4b 68 56 6a 6b 45 24 1d 1 7 52 35 53 64 1b 75 75 32 2a 4b 23 2 66 48 5 78 6a 17 6c 4e 60 5a 5d 16 12 32 49 50 57 45 52 3c 47 3c 26 44 25 a 59 3e 5f 46 1c 68 14 1e 1d 55 5f 4d 1c 20 2d 71 60 6b 72 33 13 59 67 1f 76 26 35 29 25 52 c 2 0 58 f 2 35 70 54 5d 2c 59 44 63 33 3b 1d b 7c 58 48 19 45 5a 3e 2c 19 7c 36 40 53 7b 69 17 68 5c 58 65 10 7d 2d 21 26 5b 38 6f 5f 2c 16 6b 75 42 53 4d 17 32 59 32 5e 28 32 6b 43 c 74 55 8 5e 3b 4b 68 37 1c 9 3c 53 12 6f 34 69 3 4f 8 0 3a 1b 1d 22 17 8 3 3b 7b 28 52 1e 1f 52 29 26 1d 50 4d f 54 d 8 3c 40 14 35 46 49 69 27 59 f 8 2f 10 30 54 73 32 8 26 2c 62 1b 26 38 22 2c 19 5a 20 71 70 46 52 59 1 66 d 58 39 9 16 6d 12 2 3d 2d 4b 45 70 60 20 52 61 6b 26 3c 65 79 2e 16 6d 24 53 3e 54 b 37 66 57 30 44 1 37 c 4a 49 15 27 17 27 13 73 2e 10 10 58 5 e 46 71 e 47 54 44 7f 5f 31 6f 21 2c 5a 5f 76 20 20 57 20 22 75 2 77 0 9 36 25 a 25 66 6c 5e 76 3c 2a 5e 4a 4b 46 66 3a 6 41 7e 38 70 d 21 74 7f 74 70 60 19 3c 4 5 50 4f 58 1e 58 f 29 20 5e 35 a 2 5d 6d 7e 34 75 7c 35 7c 3a 1f 5d 7a 4b 0 2b 38 20 33 57 2 5b c 2e 62 3 49 50 60 53 6f 39 f 39 45 57 5a 45 16 20 58 55 3c 5e 56 52 2b 3a 35 6e 6b 49 64 6b 52 5e 55 5b 41 56 50 11 4b 5e 78 1 67 4e 3f 8 24 47 53 4c 6c 6c 16 19 33 6f 37 75 64 56 1a 78 b 24 65 78 72 57 4 48 5c 2f 15 1d 5 54 67 1a 68 66 79 8 77 2d 46 53 77 65 6f 2e 4f 5a 50 38 46 1b 44 6d 3d 53 50 4a 66 29 6d 62 6b 4 28 58 4 46 3f 4 55 6b 7 5c 43 4 10 2f d 3a 19 74 76 12 3c 5 62 52 22 66 e 19 e 56 1c 1a 58 1e 34 55 5a 34 28 2f 59 2d 64 1 56 20 2a 5c 13 3e 17 f 23 68 6f 43 39 51 19 6c 67 68 2f 72 52 5d 5b 65 52 c 3b 2c 68 1f 79 27 1e 3d 43 2b 62 5f 45 4a 5f 3d 5b 61 51 2 13 3d 4c 56 12 37 6 3a 5b 17 45 7a 1a 37 5f 1c 16 30 29 23 3b 5 3a 28 34 7b d 7c 47 d 77 55 5f 1a 56 5c 31 73 c 5e 2b 4 a 58 7c 1d 45 10 64 4a c 4 d 25 16 23 51 3a 5e 68 69 40 9 3 1 15 8 13 7e 46 40 50 75 10 4f 6e 7 6a 2b 6d e 3d 9 1d 22 17 34 65 18 60 7e 42 37 25 10 4a 23 78]>
     but: was <[61 5e 1 70 18 8 5f 6 e 77 14 77 4d 22 26 4c 63 2 1e 5a 58 75 2c 2b 16 5d 3 e 4f b 76 51 40 1d 39 20 2c 27 52 15 8 1 58 9 3a 5d 20 3f 3c 33 22 4c 1b 25 4 e 3f 2f 66 63 43 70 53 c 1c 5a 78 27 8 7c 43 3d f 2 12 62 79 71 b 32 45 1b 2 60 2 0 19 39 2 3 32 1c 14 6b 11 a 4f 4f c 37 1e 36 37 67 12 3c 6e 17 3e 10 2d 1f 7f 4e 7e a 9 26 5 26 49 2a 7f e 76 67 7c 47 2 1d 73 0 3c 5a 3a 3e 4d 12 34 3f 1a 10 17 12 27 6d 4e 40 6a 1 5b 77 7c 5e 4c 24 79 34 3c 48 15 c 33 27 6e 32 3c 44 7f 56 3c 12 27 31 f 15 1c 7d 31 24 3c 37 5e 20 2c 32 43 31 20 7b 62 3d 7c 7d 44 6 68 35 32 29 11 34 22 1c 63 4 7e 76 73 4d 44 4d 42 5a 39 5a 78 60 d 1b 61 6a 6c 4e 24 1b 13 66 79 1b 18 50 3 52 2d 4b 52 f 1e 21 d 73 0 47 11 1b 59 43 7 43 66 5f 1e 36 31 1a 4b 14 1 6e 3 2b 4c 72 22 19 57 15 20 15 5f 36 39 d 54 37 46 41 4e 6a 34 38 2d 2a 7b 73 b 3d 3e 73 41 7e 47 10 67 21 7e 6f 2f 30 3e 72 3c 50 79 14 f 41 5c 7b c 44 17 24 2f 9 6e 16 53 23 c 15 3a 56 54 1d e 6e 21 59 51 5e 4d 73 5c 7e 47 32 5a 9 66 5 28 4e 6f 11 7f 45 e 29 41 46 66 69 5 47 4b 3f 42 7 65 6f 3 14 4c 1 8 d 55 63 62 14 50 2f 42 8 4d 2a 2f 6d 37 19 4a 3c 65 7b 4f 3 41 1a 7e 64 20 12 7b 28 2b 49 6e 18 7e d 6 e 20 54 11 5a 67 70 1f 1 79 53 22 46 76 28 35 3a 65 2 4 74 42 47 18 a 43 3d 79 60 6a 73 23 4e 56 2 1 56 65 62 34 a 25 2d 4a 74 39 15 5f 59 48 11 3c 1d 56 3e 39 37 7b 2 72 30 4e 51 0 45 6 48 21 b 54 26 32 64 20 43 2f 55 4e 10 61 4c 6f 73 4 57 5e 23 51 f 56 b 25 62 17 48 4c e 19 62 10 2c 29 1f 60 51 1f 3c 3b 7d 3f 13 64 53 5d 3b 71 3f 1e 2b 70 22 1b 4 1a 9 d 17 5b f 50 3e 5e 3e 52 76 35 7f 62 1b 2b 35 46 50 26 3a 47 68 7d 1b 7e a 2b 58 11 46 6b 4 2a 70 3 7c 2b 17 5e 33 7 5d 32 5d 62 7 2d 10 3 25 74 1c 8 2d 4d 37 5c 31 2e 0 6f c 6f 0 59 7 18 0 17 1f 78 76 47 5b 5a 1c c 4c 5 1e 1c 61 46 53 11 52 36 1d 41 5b 39 14 6b b 3b 6f 29 8 4e 2c 1e 37 6e 6a 23 7c 4b 59 27 5a 2d 72 68 3d 42 1c 42 17 1a 3b 15 16 49 1d 20 69 7f 5b 32 79 39 7c 6b 54 2b 74 2 69 15 7f 6a 6 38 42 8 70 11 2 12 3e 12 65 3d 6 3a 6d 4e 64 24 3b 34 50 a b 38 38 61 20 4c 29 6f 29 33 a e 13 24 46 20 2a 7c c 37 b 2a 67 7d 6a 9 e 69 46 6f 7e e 6 3d c 6 73 5a 0 71 51 7d 44 7c f 69 1c 31 6b 39 2e 26 31 4b 50 7c d 62 13 48 51 e 62 2c 55 48 a 25 2a 2a 7d 31 33 4a 22 77 3e 65 7e 61 3e 3c 26 64 14 10 2b 62 36 7d e 2 32 70 32 15 d 7b 4d 22 38 1c 1f 3 24 7f 7 57 3d 6b 50 27 53 5c 51 13 43 53 6c 22 63 2f 24 6f 61 8 57 64 43 40 5c 11 54 4b 75 38 1c 7 13 62 46 20 29 44 4f 54 50 40 7d 32 1d 42 f 6f 57 4b 39 69 1f 38 1d 54 21 1b 51 30 60 49 2d 70 1b 1e 23 58 4d 5c 7e 47 43 1d 74 19 4c 2c 23 3c 20 70 65 77 40 53 56 25 74 33 50 51 68 75 53 e 43 2b 7f 63 18 42 71 67 c 5c 19 7c 22 3 3d 21 60 d 76 a 30 31 3e 3c 13 6e 4b 7d 7d 53 71 2d 72 71 65 38 7b 10 7b 19 53 3f 46 59 53 4e 2c 23 2a 3c 1a 1d 20 10 67 14 5a 52 5a 10 4d 10 40 57 1b 8 26 63 6d 3d 25 23 18 6d 1e 0 58 60 39 51 58 6a 77 24 13 22 38 1f 4d 57 37 6e 5d 4c 6d 38 6f 46 7f 4b 7f 6a 7f 20 12 77 20 63 2e 1d 10 9 23 42 13 0 55 2a 79 1e 73 1c 71 1b 1c 34 20 6d 6 23 9 4f 22 7f 28 6b 41 3 32 19 75 53 27 57 66 31 59 8 1 73 6 5d 6 1b 1b 2e 42 53 26 59 38 46 7c 39 27 6a 31 2e 4a 1a 23 12 63 6c 1e 55 66 45 1b 5c 15 4d 11 58 0 41 7d 4e 6 67 47 28 18 3b 68 5f 13 31 59 7 16 63 4e 21 75 4f 35 39 69 52 3b 4 4c 39 6b 32 20 33 75 22 23 14 50 e 44 16 48 2c 46 26 d 7 34 4b 3a 3c 42 49 4c d 22 47 8 3 5e 4b 6a 63 3b f 71 34 58 5a 1e 74 32 22 5e 19 6d 6e 3a 60 61 24 23 30 56 63 75 6 5 2a 42 29 18 5 56 5c 68 1e 1a 5f 1c 41 2 60 68 76 43 43 27 12 50 65 79 6c 54 25 51 72 35 57 44 11 54 54 30 70 57 5e 2a 6e 5b 5d 37 70 6b 5 b 65 12 41 3d 4c f 3e 2d 39 a 4e 26 28 7f 3b 3a 58 30 77 52 51 14 7d 0 54 2 66 4c 46 30 14 61 46 1a 18 52 2 9 2b 1b c 3f 59 28 7e 33 31 32 7c 66 70 39 5c 19 67 32 11 42 7d 73 22 79 6e 44 11 66 29 2c 7 4b 7e 5c f 37 64 16 33 8 7d 78 77 74 12 b 50 7 34 1 8 7b 79 34 21 36 56 79 48 58 6 18 7a 70 41 31 3d 3a 15 58 7 20 7f 24 54 13 51 1c 58 3e 9 3f 2d 42 66 13 32 54 17 58 f 37 51 7f 6a 2 3a e 4 47 75 31 68 71 60 59 1e 1e 46 40 50 6c 7a 3a 1d 76 21 6f 3e 34 11 55 30 42 64 11 4c 5f 30 70 17 3 28 26 5b 33 7d 68 4 19 75 6e 2 69 66 7c 5b 79 76 69 4b 6b 50 0 b 7f 57 14 25 5c 7e c 22 8 32 0 28 36 31 27 50 f 45 29 7b 41 66 1c 1f 64 2f 2d 28 26 29 1c 5d 63 6d 7b 2b 4a 6f 4f 66 25 0 4c 16 7e 37 e 62 2c 5c 73 62 52 7f 4e 2a 30 4b 30 39 12 6b 27 77 51 1a 39 79 64 68 42 60 60 2d 5a 59 6c 4f 11 63 1f 53 66 3c 36 43 39 42 78 5b 7 24 46 7c 5b 7b 71 40 75 7a 3c 55 13 27 62 74 37 49 65 12 24 66 3 8 12 6c 64 11 56 41 7a 38 4c 1b f 67 25 14 74 1f 55 1b 4f 9 e 60 4f 46 f 42 20 4b 9 51 48 43 42 71 4a 4d 7a 23 30 53 44 41 1 2d 7d 4d 58 15 33 2d 30 57 56 41 28 a 8 36 5b 31 75 38 61 1c 6c 5c 49 11 75 6c 18 f 48 6a b 37 2d 4b 30 5e 19 21 59 7c 7a 16 18 b 4f 3f 76 6d 28 16 33 55 57 31 16 1d 7 51 34 48 7e 6 52 39 6c 34 d 44 5e 65 2e 41 21 40 7b 5a 78 39 30 76 5b 12 11 3d 59 51 18 23 5e 2f 44 57 9 6b 2a 17 a d 79 4 35 7b 38 6a 7a 3e 2 3 18 d 48 7b 40 35 63 76 3f 41 4d 66 0 a 7d 1 59 58 5 65 30 1 77 2f 4 5d 62 5d 6b 39 6a 18 5e 4b 67 0 11 0 65 6b 4b 36 1e 6d 66 0 78 71 3e 22 5 17 74 41 9 1a 73 50 3c 12 48 39 66 7e 4b 68 56 6a 6b 45 24 1d 1 7 52 35 53 64 1b 75 75 32 2a 4b 23 2 66 48 5 78 6a 17 6c 4e 60 5a 5d 16 12 32 49 50 57 45 52 3c 47 3c 26 44 25 a 59 3e 5f 46 1c 68 14 1e 1d 55 5f 4d 1c 20 2d 71 60 6b 72 33 13 59 67 1f 76 26 35 29 25 52 c 2 0 58 f 2 35 70 54 5d 2c 59 44 63 33 3b 1d b 7c 58 48 19 45 5a 3e 2c 19 7c 36 40 53 7b 69 17 68 5c 58 65 10 7d 2d 21 26 5b 38 6f 5f 2c 16 6b 75 42 53 4d 17 32 59 32 5e 28 32 6b 43 c 74 55 8 5e 3b 4b 68 37 1c 9 3c 53 12 6f 34 69 3 4f 8 0 3a 1b 1d 22 17 8 3 3b 7b 28 52 1e 1f 52 29 26 1d 50 4d f 54 d 8 3c 40 14 35 46 49 69 27 59 f 8 2f 10 30 54 73 32 8 26 2c 62 1b 26 38 22 2c 19 5a 20 71 70 46 52 59 1 66 d 58 39 9 16 6d 12 2 3d 2d 4b 45 70 60 20 52 61 6b 26 3c 65 79 2e 16 6d 24 53 3e 54 b 37 66 57 30 44 1 37 c 4a 49 15 27 17 27 13 73 2e 10 10 58 5 e 46 71 e 47 54 44 7f 5f 31 6f 21 2c 5a 5f 76 20 20 57 20 22 75 2 77 0 9 36 25 a 25 66 6c 5e 76 3c 2a 5e 4a 4b 46 66 3a 6 41 7e 38 70 d 21 74 7f 74 70 60 19 3c 4 5 50 4f 58 1e 58 f 29 20 5e 35 a 2 5d 6d 7e 34 75 7c 35 7c 3a 1f 5d 7a 4b 0 2b 38 20 33 57 2 5b c 2e 62 3 49 50 60 53 6f 39 f 39 45 57 5a 45 16 20 58 55 3c 5e 56 52 2b 3a 35 6e 6b 49 64 6b 52 5e 55 5b 41 56 50 11 4b 5e 78 1 67 4e 3f 8 24 47 53 4c 6c 6c 16 19 33 6f 37 75 64 56 1a 78 b 24 65 78 72 57 4 48 5c 2f 15 1d 5 54 67 1a 68 66 79 8 77 2d 46 53 77 65 6f 2e 4f 5a 50 38 46 1b 44 6d 3d 53 50 4a 66 29 6d 62 6b 4 28 58 4 46 3f 4 55 6b 7 5c 43 4 10 2f d 3a 19 74 76 12 3c 5 62 52 22 66 e 19 e 56 1c 1a 58 1e 34 55 5a 34 28 2f 59 2d 64 1 56 20 2a 5c 13 3e 17 f 23 68 6f 43 39 51 19 6c 67 68 2f 72 52 5d 5b 65 52 c 3b 2c 68 1f 79 27 1e 3d 43 2b 62 5f 45 4a 5f 3d 5b 61 51 2 13 3d 4c 56 12 37 6 3a 5b 17 45 7a 1a 37 5f 1c 16 30 29 23 3b 5 3a 28 34 7b d 7c 47 d 77 55 5f 1a 56 5c 31 73 c 5e 2b 4 a 58 7c 1d 45 10 64 4a c 4 d 25 16 23 51 3a 5e 68 69 40 9 3 1 15 8 13 7e 46 40 50 75 10 4f 6e 7 6a 2b 6d e 3d 9 1d 22 17 34 65 18 60 7e 42 37 25 10 4a 23 78]>
	at __randomizedtesting.SeedInfo.seed([6C140CED5816D457:6FA0A1EF62A68B9]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2524)
	at org.elasticsearch.xpack.esql.expression.function.AbstractScalarFunctionTestCase.testEvaluateBlock(AbstractScalarFunctionTestCase.java:264)
	at org.elasticsearch.xpack.esql.expression.function.AbstractScalarFunctionTestCase.testCrankyEvaluateBlockWithoutNulls(AbstractScalarFunctionTestCase.java:187)
```
Also reproducible with
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.scalar.string.ReverseTests.testCrankyEvaluateBlockWithoutNulls {TestCase=<long unicode TEXT>}" -Dtests.seed=6C140CED5816D457 -Dtests.locale=bho-IN -Dtests.timezone=Europe/Kiev -Druntime.java=22
```

--- a/docs/changelog/115245.yaml
+++ b/docs/changelog/115245.yaml
@@ -1,0 +1,7 @@
+pr: 115245
+summary: "ESQL: Fix `REVERSE` with backspace character"
+area: ES|QL
+type: bug
+issues:
+ - 115227
+ - 115228

--- a/docs/changelog/115245.yaml
+++ b/docs/changelog/115245.yaml
@@ -3,5 +3,6 @@ summary: "ESQL: Fix `REVERSE` with backspace character"
 area: ES|QL
 type: bug
 issues:
+ - 114372
  - 115227
  - 115228

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -346,12 +346,8 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/115129
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
   issue: https://github.com/elastic/elasticsearch/issues/115135
-- class: org.elasticsearch.xpack.esql.expression.function.scalar.string.ReverseTests
-  method: testEvaluateInManyThreads {TestCase=<long unicode TEXT>}
-  issue: https://github.com/elastic/elasticsearch/issues/115227
-- class: org.elasticsearch.xpack.esql.expression.function.scalar.string.ReverseTests
-  method: testEvaluateInManyThreads {TestCase=<long unicode KEYWORD>}
-  issue: https://github.com/elastic/elasticsearch/issues/115228
+- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
+  issue: https://github.com/elastic/elasticsearch/issues/115213
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=esql/60_usage/Basic ESQL usage output (telemetry)}
   issue: https://github.com/elastic/elasticsearch/issues/115231

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -176,9 +176,6 @@ tests:
 - class: org.elasticsearch.packaging.test.WindowsServiceTests
   method: test81JavaOptsInJvmOptions
   issue: https://github.com/elastic/elasticsearch/issues/113313
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=mtermvectors/10_basic/Tests catching other exceptions per item}
-  issue: https://github.com/elastic/elasticsearch/issues/113325
 - class: org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT
   method: test {yaml=reference/ccr/apis/follow/post-resume-follow/line_84}
   issue: https://github.com/elastic/elasticsearch/issues/113343

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -176,6 +176,9 @@ tests:
 - class: org.elasticsearch.packaging.test.WindowsServiceTests
   method: test81JavaOptsInJvmOptions
   issue: https://github.com/elastic/elasticsearch/issues/113313
+- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
+  method: test {p0=mtermvectors/10_basic/Tests catching other exceptions per item}
+  issue: https://github.com/elastic/elasticsearch/issues/113325
 - class: org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT
   method: test {yaml=reference/ccr/apis/follow/post-resume-follow/line_84}
   issue: https://github.com/elastic/elasticsearch/issues/113343
@@ -343,8 +346,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/115129
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
   issue: https://github.com/elastic/elasticsearch/issues/115135
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  issue: https://github.com/elastic/elasticsearch/issues/115213
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=esql/60_usage/Basic ESQL usage output (telemetry)}
   issue: https://github.com/elastic/elasticsearch/issues/115231

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Reverse.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Reverse.java
@@ -10,7 +10,6 @@ package org.elasticsearch.xpack.esql.expression.function.scalar.string;
 import org.apache.lucene.util.BytesRef;
 import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
 import org.elasticsearch.common.io.stream.StreamInput;
-import org.elasticsearch.common.lucene.BytesRefs;
 import org.elasticsearch.compute.ann.Evaluator;
 import org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator;
 import org.elasticsearch.xpack.esql.core.expression.Expression;
@@ -79,8 +78,6 @@ public class Reverse extends UnaryScalarFunction {
 
     /**
      * Reverses a unicode string, keeping grapheme clusters together
-     * @param str
-     * @return
      */
     public static String reverseStringWithUnicodeCharacters(String str) {
         BreakIterator boundary = BreakIterator.getCharacterInstance(Locale.ROOT);
@@ -100,10 +97,12 @@ public class Reverse extends UnaryScalarFunction {
         return reversed.toString();
     }
 
-    private static boolean isOneByteUTF8(BytesRef ref) {
+    private static boolean reverseBytesIsReverseUnicode(BytesRef ref) {
         int end = ref.offset + ref.length;
         for (int i = ref.offset; i < end; i++) {
-            if (ref.bytes[i] < 0) {
+            if (ref.bytes[i] < 0 // Anything encoded in multibyte utf-8
+                || ref.bytes[i] == 0x28 // Backspace
+            ) {
                 return false;
             }
         }
@@ -112,13 +111,13 @@ public class Reverse extends UnaryScalarFunction {
 
     @Evaluator
     static BytesRef process(BytesRef val) {
-        if (isOneByteUTF8(val)) {
+        if (reverseBytesIsReverseUnicode(val)) {
             // this is the fast path. we know we can just reverse the bytes.
             BytesRef reversed = BytesRef.deepCopyOf(val);
             reverseArray(reversed.bytes, reversed.offset, reversed.length);
             return reversed;
         }
-        return BytesRefs.toBytesRef(reverseStringWithUnicodeCharacters(val.utf8ToString()));
+        return new BytesRef(reverseStringWithUnicodeCharacters(val.utf8ToString()));
     }
 
     @Override

